open-sphere-base/analysis/src/main/java/io/opensphere/analysis/listtool/model/DataElementProvider.java

```diff
@@ -77,16 +81,18 @@
      * Constructor.
      *
      * @param model the table model
-     * @param mantleToolbox the mantle toolbox
+     * @param toolbox the toolbox
      * @param dataType the data type
      * @param metaColumns the meta columns
      * @param timeColumnIndex the time column index
      */
-    public DataElementProvider(TableModel model, MantleToolbox mantleToolbox, DataTypeInfo dataType,
-            List<MetaColumn<?>> metaColumns, int timeColumnIndex)
+    public DataElementProvider(TableModel model, Toolbox toolbox, DataTypeInfo dataType, List<MetaColumn<?>> metaColumns,
+            int timeColumnIndex)
     {
         super(model);
-        if (mantleToolbox != null)
+        myToolbox = toolbox;
+        MantleToolbox mantleToolbox;
+        if (toolbox != null && (mantleToolbox = MantleToolboxUtils.getMantleToolbox(toolbox)) != null)

```

Comment:
I would argue that toolbox should never be null, and should be checked as a precondition instead of being defensive here. And, as long as toolbox isn't null, it should be safe to assume that MantleToolbox is not null either.

Reply:
Good point. Fixed.